### PR TITLE
Implement S3 Bucket Suitable for Serving Static Websites on S3

### PR DESCRIPTION
**Pull request by osterman**
## what
* Create S3 bucket
* Enable versioning
* Support log storage to S3 bucket
* Support `A` record `ALIAS` to bucket name
* Support CORS
* Support Bucket Redirects
* Support routing rules

## why
* Efficient & Affordable hosting of static sites on S3
* Use as upstream for CloudFront distributions that require more advanced configurations

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    allowed_origins = "${var.cors_allowed_origins}"
+    expose_headers  = "${var.cors_expose_headers}"
+    max_age_seconds = "${var.cors_max_age_seconds}"
+  }
+
+  versioning {
+    enabled = "${var.versioning_enabled}"
+  }
+
+  lifecycle_rule {
```
Comment (SweetOps) [inline main.tf]: I don't sure about `lifecycle_rule ` for website bucket. Could you explain for what it need?

Comment (osterman) [reply main.tf]: This is so we can have versioned content in the bucket. The lifecycle rule manages how long we keep those old versions.

**Comment by SweetOps** [comment]
> But in production, it's better to use CDN.

@aknysh , for this case we've `tf_cdn`

**Comment by aknysh** [comment]
OK I see that we have `tf_cdn_s3` for this use-case.
So `tf_s3_website` is for simple websites hosted on `S3` without `CDN`.

**Comment by comeanother** [comment]
Tested.
screenshot from 2017-09-19 10-30-46

Content is accessible.
screenshot from 2017-09-19 10-32-50

**Comment by osterman** [comment]
@aknysh it's also for sites that require [`routing_rules`](https://www.terraform.io/docs/providers/aws/r/s3_bucket.html#routing_rules) which are not supported by vanilla S3 buckets. We have current use-case where we need to redirect an object to a versioned object (e.g. `/foo.js` -> `/foo.1234.js`). This kind of redirect requires an S3 website.

**Pull request merged**